src/goGenerateTests.ts

```diff
@@ -83,32 +83,30 @@ export function generateTestCurrentFile(): Thenable<boolean> {
 		vscode.workspace.getConfiguration('go', editor.document.uri));
 }
 
-export function generateTestCurrentFunction(): Thenable<boolean> {
-	let editor = checkActiveEditor();
+export async function generateTestCurrentFunction(): Promise<boolean> {
+	const editor = checkActiveEditor();
 	if (!editor) {
 		return;
 	}
 
-	return getFunctions(editor.document).then(functions => {
-		let currentFunction: vscode.SymbolInformation;
-		for (let func of functions) {
-			let selection = editor.selection;
-			if (selection && func.location.range.contains(selection.start)) {
-				currentFunction = func;
-				break;
-			}
-		}
-		if (!currentFunction) {
-			vscode.window.showInformationMessage('No function found at cursor.');
-			return Promise.resolve(false);
-		}
-		let funcName = currentFunction.name;
-		if (funcName.includes('.')) {
-			funcName = funcName.split('.')[1];
+	const functions = await getFunctions(editor.document);
+	const selection = editor.selection;
+	const currentFunction: vscode.DocumentSymbol = functions.find(func => {
+		if (selection && func.range.contains(selection.start)) {
+			return true;
 		}
-		return generateTests({ dir: editor.document.uri.fsPath, func: funcName },
-			vscode.workspace.getConfiguration('go', editor.document.uri));
+		return false;
```

Comment:
The above can be simplified to `return func.range.contains(selection.start)`
Active editors will always have a selection